Makefile

```diff
@@ -1,41 +1,40 @@
 # $Header: /home/henk/CVS/dealer/Makefile,v 1.15 1999/08/05 19:57:44 henk Exp $
 
 CC      = gcc
-CFLAGS = -Wall -pedantic -O2 -I. -DNDEBUG -c $(shell dpkg-buildflags --get CFLAGS) $(shell dpkg-buildflags --get CPPFLAGS)
+CFLAGS = -Wall -pedantic -I. -DNDEBUG -c
 FLEX    = flex
 YACC    = yacc
+# Note: this should be the Berkeley Yacc, sometimes called byacc
 
 PROGRAM  = dealer
 TARFILE  = ${PROGRAM}.tar
 GZIPFILE = ${PROGRAM}.tar.gz
 
-SRC  = dealer.c pbn.c  c4.c getopt.c pointcount.c
+SRC  = dealer.c pbn.c  c4.c getopt.c pointcount.c __random.c rand.c srand.c
 LSRC = scan.l
 YSRC = defs.y
 HDR  = dealer.h tree.h
 
-OBJ  = dealer.o defs.o pbn.o c4.o getopt.o pointcount.o
+OBJ  = dealer.o defs.o pbn.o c4.o getopt.o pointcount.o __random.o rand.o srand.o
 LOBJ = scan.c
 YOBJ = defs.c
-MAN  = dealer.6 dealer.dpp.6
 
 
-dealer: ${OBJ} ${LOBJ} ${YOBJ} ${MAN}
-	${MAKE} -C Random lib
-	$(CC) -o $@ ${OBJ} -L./Random -lgnurand $(shell dpkg-buildflags --get LDFLAGS)
-
+dealer: ${OBJ} ${LOBJ} ${YOBJ}
+	$(CC) -o $@ ${OBJ} 
+	
 clean:
-	rm -f ${OBJ} ${LOBJ} ${YOBJ} ${MAN} dealer
+	rm -f ${OBJ} ${LOBJ} ${YOBJ} 
 	${MAKE} -C Examples clean
-	${MAKE} -C Random   clean
 
 tarclean: clean ${YOBJ}
-	rm -f dealer
+	rm -f ${PROGRAM}
 	rm -f ${TARFILE}  ${GZIPFILE}
 
 tarfile: tarclean
 	cd .. ; \
-	tar cvf ${TARFILE} ${PROGRAM} --exclude CVS --exclude ${TARFILE}; \
+        rm ${TARFILE} ${GZIPFILE} ; \
+	tar cvf ${TARFILE} ${PROGRAM} ; \
```

Comment:
Hold up. The `tarfile` target depends on the `tarclean` target, which deletes `${PROGRAM}`. How does this succeed?

...and maybe the answer is in the `cd ..` 3 lines up: perhaps the `tarclean` tries to `rm -f` the program in the wrong directory and silently does nothing when it's not there? I dunno, something in here needs cleanup.

Reply:
The `${PROGRAM}` on the current line doesn't refer to `dealer` the executable built by this makefile; it refers to the `dealer/` directory in which the makefile resides. I'll rephrase so that `${PROGRAM}` always refers to a program.